src/prog_models/prognostics_model.py

```diff
@@ -1147,7 +1147,9 @@ def generate_surrogate(self, load_functions, first_output = None, threshold_keys
             'inputs': self.inputs,
             'outputs': self.outputs,
             'events': self.events,
-            'stability_tol': 1e-05
+            'stability_tol': 1e-05,
+            'data_add_noise': True,
```

Comment:
```suggestion
            'data_add_noise': True,
```

Rather than having 2 parameters, what if we just have one (magnitude) and if it's set to 0 there is no noise?